src/Drush/Traits/SiteUriTrait.php

```diff
@@ -53,4 +55,32 @@ private function getSitesSubdirFromUri(string $root, string $uri): mixed {
     return FALSE;
   }
 
+  /**
+   * Preparing site uri if --uri="https://www.example.com"
+   *
+   * @param string $uri
+   *   Site uri in a url format.
+   *
+   * @return string
+   *   Site uri with format port.domain.path or normal uri.
+   */
+  protected function prepareSiteUri(string $uri): string {
+    if (filter_var($uri, FILTER_VALIDATE_URL) !== FALSE) {
+      $parseUrl = parse_url($uri);
+      $siteUri = "";
+      if (isset($parseUrl['port'])) {
+        $siteUri = $parseUrl['port'] . ".";
+      }
+      if (isset($parseUrl['host'])) {
+        $siteUri = $siteUri . $parseUrl['host'];
+      }
+      if (isset($parseUrl['path'])) {
+        $siteUri = $siteUri . str_replace("/", ".", rtrim($parseUrl['path'], "/"));
+      }
+      return trim($siteUri);
+    }
+
+    return $uri;
+  }
+
```

Comment:
Why we have added this method here, though we already have `getSitesSubdirFromUri()` which do the same thing. we have copied this function from drush itself

Reply:
Verified changes added following snippet in sites.php,  and I don't see any issue.
```
$sites = [
  'example.com' => 'example',
  'dev.drupal.org' => 'example',
  'localhost.example' => 'example',
  '8080.localhost.example' => 'example',
  '8080.www.drupal.org.my-site.test' => 'example'
];
```
Then ran site install with following set of uri:
-  http://dev.drupal.org
-  http://localhost/example
-  http://localhost:8080/example
- https://www.drupal.org:8080/my-site/test